addons/l10n_ro_edi/models/ciusro_document.py

```diff
@@ -0,0 +1,198 @@
+
+        return response
+
+    def _request_ciusro_send_invoice(self, xml_data):
+        self.ensure_one()
+        response = self._make_request('upload', xml_data)
+        if not response:
+            return
+
+        root = etree.fromstring(response.content)
+        res_status = root.get('ExecutionStatus')
+        if res_status == '1':
+            error_elements = root.findall('.//ns:Errors', namespaces=NS_UPLOAD)
+            error_messages = [error_element.get('errorMessage') for error_element in error_elements]
+            return self._handle_error('\n'.join(error_messages))
+        else:
+            self.state = 'sending'
+            self.key_loading = root.get('index_incarcare')
+
+    def _request_ciusro_fetch_status(self, session=None):
+        self.ensure_one()
+        response = self._make_request('stareMesaj', session=session)
+        if not response:
+            return
+
+        root = etree.fromstring(response.content)
+        error_elements = root.findall('.//ns:Errors', namespaces=NS_STATUS)
+        if error_elements:
+            error_messages = [error_element.get('errorMessage') for error_element in error_elements]
+            return self._handle_error('\n'.join(error_messages))
+
+        state_status = root.get('stare')
+        self.key_download = root.get('id_descarcare')
+        if state_status in ('nok', 'ok'):
+            self.state = state_status
+
+    def _request_ciusro_download_answer(self):
+        self.ensure_one()
+        response = self._make_request('descarcare')
+        if not response:
+            return
+
+        # E-Factura gives download response in ZIP format
+        zip_ref = zipfile.ZipFile(io.BytesIO(response.content))
+        signature_file = next(file for file in zip_ref.namelist() if 'semnatura' in file)
+
+        xml_content = zip_ref.open(signature_file)
+        root = etree.parse(xml_content)
+        error_element = root.find('.//ns:Error', namespaces=NS_HEADER)
+        self.key_signature = root.find('.//ns:SignatureValue', namespaces=NS_SIGNATURE).text
+        self.key_certificate = root.find('.//ns:X509Certificate', namespaces=NS_SIGNATURE).text
+        if error_element is not None:
+            error_message = error_element.get('errorMessage')
+            return self._handle_error(error_message)
+
+        self.env['ir.attachment'].create({
+            'name': self._get_attachment_file_name(),
+            'raw': response.content,
+            'mimetype': 'application/zip',
+            'res_model': self._name,
+            'res_id': self.id,
+            'res_field': 'attachment_id',
+        })
+        self.invalidate_recordset(fnames=['attachment_id'])
+        self.state = 'sent'
+        # Delete all previous error documents
+        to_delete_documents = self.env['ciusro.document'].filtered(lambda d: d.move_id == self.move_id and d.id != self.id)
+        to_delete_documents.unlink()
```

Comment:
You don't delete the previously created attachments here.

Instead of creating a document and then, making it evolving other the time, just trash the previously failed ones, then create a new one depending if the web-service successed or not (all those methods are specific to invoices so let's put them on account.move instead).